extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java

```diff
@@ -52,12 +54,9 @@
 import io.quarkus.smallrye.health.deployment.spi.HealthBuildItem;
 
 public class MongoClientProcessor {
-    private static final DotName MONGOCLIENT_ANNOTATION = DotName.createSimple(MongoClientName.class.getName());
-
-    @BuildStep
-    BeanDefiningAnnotationBuildItem registerConnectionBean() {
-        return new BeanDefiningAnnotationBuildItem(MONGOCLIENT_ANNOTATION);
-    }
+    private static final DotName MONGO_CLIENT_ANNOTATION = DotName.createSimple(MongoClientName.class.getName());
```

Comment:
If it didn't work before, please let's move this annotation to the API package i.e. no `runtime.`. It's user API.

Also we will probably need to update the doc and quickstart if it's mentioned.

Reply:
Agreed to do this in a follow up